workflows/gtfs_validator_execution.yml

```diff
@@ -0,0 +1,187 @@
+main:
+  params: [event]
+  steps:
+      - log_event:
+          call: sys.log
+          args:
+              text: ${event}
+              severity: INFO
+      - extract_data:
+          assign:
+          - data: ${event.data.protoPayload}
+      - init:
+          assign:
+            - datasetsBucket: ${sys.get_env("datasets_bucket_name")}
+            - reportsBucket: ${sys.get_env("reports_bucket_name")}
+            - validatorEndpoint: ${sys.get_env("validator_endpoint")}
+            - feedID: ${text.split(data.resourceName, "/")[5]}
+            - datasetID: ${text.split(data.resourceName, "/")[6]}
+            - url: ${"https://storage.googleapis.com/" + datasetsBucket + "/" + feedID + "/" + datasetID + "/" + datasetID + ".zip"}
+            - endpoint: ${validatorEndpoint + "/create-job"}
+            - payload:
+                  url: "${url}"
+                  countryCode: ""
+            - headers:
+                  Content-Type: application/json
+            - jobID: "undefined"
+      - assignResultsEndpoint:
+          switch:
+            - condition: ${sys.get_env("environment") == "prod"}
+              next: assignResultsEndpointProd
+          next: assignResultsEndpointDev
+      - assignResultsEndpointProd:
+          assign:
+            - resultsEndpoint: "https://gtfs-validator-results.mobilitydata.org/"
+      - assignResultsEndpointDev:
+          assign:
+            - resultsEndpoint: "https://staging-gtfs-validator-results.mobilitydata.org/"
+      - logInit:
+          call: sys.log
+          args:
+              text: '${"feedID: " + feedID + ", datasetID: " + datasetID + ", url: " + url}'
+              severity: INFO
+      - createJob:
+          call: http.post
+          args:
+              url: "${endpoint}"
+              headers: ${headers}
+              body: ${payload}
+          result: response
+      - checkJobCreationStatus:
+          switch:
+            - condition: ${response.code == 200}
+              next: successCreatingJob
+          next: errorCreatingJob
+      - errorCreatingJob:
+          return:
+            createJobResponse: response
+      - successCreatingJob:
+          assign:
+              - jobID: ${response.body.jobId}
+      - logJobID:
+          call: sys.log
+          args:
+              text: '${"jobID: " + jobID}'
+              severity: INFO
+      - waitForExecutionResultCreation:
```

Comment:
My friend copilot tells me there is a retry mechanism you might be able to use.
```
- waitForExecutionResultCreation:
    retry:
      maxAttempts: 120
      delay: 10s
    steps:
      ...

```
If it does work, it looks simpler that the looping used here. Maybe you can consider it?

Reply:
It ended up being slightly more complicated that what your friend estimated, but it's done ✅